source/architecture-partials/architecture.md

```diff
@@ -0,0 +1,100 @@
+
+3. **Production** Promoting code to production is a manual step: use the "Promote to production..." button in the 
+   Heroku pipeline.  This will immediately release the master branch code to production, with no further test run.
+
+## Static site build
+
+The Publisher app allows users to create, update, edit and preview pages for the Ethnicity Facts and Figures site. 
+For these changes to be reflected in the pages of the live site they need to be built and pushed to the production S3
+bucket. A build of the entire site is requested by Publisher whenever a page is marked as published (or, in rare cases,
+unpublished). The request for a build is stored in the `build` table in the database.
+ 
+The static build is run by a Heroku scheduled task which spins up a new instance of the Publisher application within the
+environment (which is not accessible on any URL) and runs the `build_static_site` management command. This command first
+checks if there are any builds pending in the `build` table, and if there are it builds the whole site into a folder and
+(optionally, depending on environment variables) copies the the entire folder to S3 and/or pushes the folder to a 
+remote Git repository.
+
+Site builds in Production go to the production bucket, and to the `ethnicity-facts-and-figures-build-history` repository
+on GitHub.  This acts as a backup of the entire site (so if S3 became unavailable for some reason it could theoretically
+be deployed elswhere), and a history of all published versions of the site.
+
+## Scheduled tasks in Heroku
+### Build management
+There are a number of tasks that run in relation to building the static site:
+
+* `build_static_site` runs every 10 minutes and checks if there are any builds pending in the database; if there are it 
+  runs the build
+* `delete_old_builds` runs overnight and clears out the build table of successfully completed builds
+* `report_broken_build` and `report_stalled_build` run hourly and will send an email to notify developers if any builds
+  have been marked as failed in the database or if any builds have been running for over half an hour (have stalled).
+
+### Database management
+The dashboard pages in the Publisher use data from materialized views, and the `refresh_materialized_views` command 
+runs nightly to keep these views up to date. (The views are also refreshed right before every static build of the site, 
+so that dashboards published on the live site are always up-to-date.)
+
+We keep the data in our Staging database up-to-date by running the `pull_prod_data` command nightly.  This copies data
+from the production database into the staging database, removing all user email addresses and passwords and setting up
+some known user accounts that can be used for testing in Staging and on review apps.
+
+## Heroku add-on services
+### Email
+We use Mailgun for sending emails from Publisher.  The only emails we send from the app are for users to confirm new 
```

Comment:
Maybe worth talking about how we've got email set up?

Should we also talk about WorkMail?

Reply:
I wasn't going to mention WorkMail, but could?

Reply:
Probably worth mentioning, as otherwise easy for that to get totally forgotten.

Reply:
Does it live here though?  It feels like a separate thing from the architecture.

Reply:
And what kind of thing would you add about how email is set up?

Reply:
I've added a note about WorkMail and Secrets Manager and updated the diagram.  See what you reckon.